csrc/multidevice/utils.cpp

```diff
@@ -227,6 +227,19 @@ void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs) {
   }
 }
 
+void shardBetween(
+    std::vector<TensorView*> ref_tvs,
```

Comment:
```suggestion
    const std::vector<TensorView*>& ref_tvs,
```
In most cases, this saves a copy or is as fast.